src/test/java/org/fogbowcloud/manager/core/processors/FulfilledProcessorTest.java

```diff
@@ -99,355 +95,365 @@ public void tearDown() {
+        Assert.assertNull(this.failedOrderList.getNext());
     }
 
-    /**
-     * Test if a fulfilled order of an active localidentity compute instance has not being changed
-     * to failed if SSH connectivity is reachable.
-     *
-     * @throws InterruptedException
-     */
+    // test case: When running thread in the FulfilledProcessor and the InstanceState is Ready, the
+    // method processFulfilledOrder() must not change OrderState to Failed and must remain in
+    // Fulfilled list.
+    @SuppressWarnings("static-access")
     @Test
     public void testRunProcessLocalComputeOrderInstanceReachable() throws Exception {
+
+        // set up
         Order order = this.createOrder();
         order.setOrderState(OrderState.FULFILLED);
         this.fulfilledOrderList.addItem(order);
+        Assert.assertNull(this.failedOrderList.getNext());
 
         Instance orderInstance = Mockito.spy(new ComputeInstance(FAKE_INSTANCE_ID));
         orderInstance.setState(InstanceState.READY);
         order.setInstanceId(FAKE_INSTANCE_ID);
 
+        // exercise
         mockCloudConnectorFactory(orderInstance);
-
         Mockito.when(this.sshConnectivity
                 .checkSSHConnectivity(Mockito.any(SshTunnelConnectionData.class))).thenReturn(true);
 
-        Assert.assertNull(this.failedOrderList.getNext());
-
         this.thread = new Thread(this.fulfilledProcessor);
         this.thread.start();
+        this.thread.sleep(DEFAULT_SLEEP_TIME);
 
-        Thread.sleep(MIN_SLEEP_TIME);
+        // verify
+        Mockito.verify(this.sshConnectivity)
+                .checkSSHConnectivity(Mockito.any(SshTunnelConnectionData.class));
 
-        Order test = this.fulfilledOrderList.getNext();
-        Assert.assertNotNull(test);
-        Assert.assertEquals(order.getInstanceId(), test.getInstanceId());
-        Assert.assertEquals(OrderState.FULFILLED, test.getOrderState());
+        Assert.assertNotNull(this.fulfilledOrderList.getNext());
+        Assert.assertNull(this.failedOrderList.getNext());
     }
 
-    /**
-     * Test if a fulfilled order of an active localidentity compute instance is changed to failed if
-     * SSH connectivity is not reachable.
-     *
-     * @throws InterruptedException
-     */
+    // test case: When running thread in the FulfilledProcessor and the InstanceState is not Ready,
+    // the processFulfilledOrder() method must change the OrderState to Failed by adding in that
+    // list, and removed from the Fulfilled list.
+    @SuppressWarnings("static-access")
     @Test
     public void testRunProcessLocalComputeOrderInstanceNotReachable() throws Exception {
+
+        // set up
         Order order = this.createOrder();
         order.setOrderState(OrderState.FULFILLED);
         this.fulfilledOrderList.addItem(order);
+        Assert.assertNull(this.failedOrderList.getNext());
 
         Instance orderInstance = Mockito.spy(new ComputeInstance(FAKE_INSTANCE_ID));
```

Comment:
Do we really need to spy this object?

Reply:
No, I will remove the spy from this object.